src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -240,114 +210,57 @@ object ScalafixPlugin extends AutoPlugin {
-
-  private val scalafixParser =
-    ScalafixCompletions.parser(workingDirectory.toPath, compat = false)
-  private val scalafixParserCompat =
-    ScalafixCompletions.parser(workingDirectory.toPath, compat = true)
-
-  def scalafixInputTask(
-      parser: Parser[Seq[String]],
-      compat: Boolean,
-      mode: ScalafixMainMode,
-      config: Configuration
-  ): Def.Initialize[InputTask[Unit]] =
-    Def.inputTaskDyn {
-      scalafixCompileTask(parser.parsed, compat, mode, config)
-    }
-
-  def scalafixCompileTask(
-      inputArgs: Seq[String],
-      compat: Boolean,
-      mode: ScalafixMainMode,
-      config: Configuration
-  ): Def.Initialize[Task[Unit]] =
-    Def.taskDyn {
-      compile.in(config).value // trigger compilation
+  def scalafixCompileTask(mode: ScalafixMainMode, config: Configuration): Def.Initialize[InputTask[Unit]] =
+    Def.inputTask {
       val scalafixClasspath =
         classDirectory.in(config).value.toPath +:
           dependencyClasspath.in(config).value.map(_.data.toPath)
+
       val sourcesToFix = for {
         source <- unmanagedSources.in(config).in(scalafix).value
         if source.exists()
         if canFix(source)
       } yield source.toPath
-      scalafixMainTask(
-        inputArgs,
-        compat,
-        mode,
-        sourcesToFix,
-        thisProject.value.id,
-        streams.value,
-        scalafixClasspath
-      )
-    }
 
-  def scalafixMainTask(
-      inputArgs: Seq[String],
-      compat: Boolean,
-      mode: ScalafixMainMode,
-      files: Seq[Path],
-      projectId: String,
-      streams: TaskStreams,
-      classpath: Seq[Path] = Nil
-  ): Def.Initialize[Task[Unit]] = {
-    if (files.isEmpty) Def.task(())
-    else {
-      Def.task {
-        val (api, baseArgs) = scalafixAPI.value
-        var mainArgs = baseArgs
-          .withMode(mode)
-          .withPaths(files.asJava)
-          .withClasspath(classpath.asJava)
-
-        if (scalafixVerbose.value) {
-          mainArgs = mainArgs.withArgs(List("--verbose").asJava)
-        }
+      val (api, baseArgs) = scalafixAPI.value
 
-        scalafixConfig.value match {
-          case Some(x) =>
-            mainArgs = mainArgs.withConfig(x.toPath)
-          case _ =>
-        }
+      var mainArgs = baseArgs
+        .withMode(mode)
+        .withPaths(sourcesToFix.asJava)
+        .withClasspath(scalafixClasspath.asJava)
+
+      if (scalafixVerbose.value) {
+        mainArgs = mainArgs.withArgs(List("--verbose").asJava)
+      }
 
-        if (compat && inputArgs.nonEmpty) {
+      scalafixConfig.value match {
+        case Some(x) =>
+          mainArgs = mainArgs.withConfig(x.toPath)
+        case _ =>
+      }
+      val logger = streams.value.log
+
+      val workingDirectory = baseDirectory.in(ThisBuild).value.toPath
+      val loadedRules = mainArgs.availableRules.asScala.toList
+
+      val inputArgs = new ScalafixCompletions(workingDirectory, loadedRules).parser.parsed
```

Comment:
@Duhemm @jvican, I would like to have some help here. I get:

```
[error] /home/gui/sbt-scalafix/src/main/scala/scalafix/sbt/ScalafixPlugin.scala:246:47: Illegal dynamic reference: workingDirectory
[error]       val inputArgs = new ScalafixCompletions(workingDirectory, loadedRules).parser.parsed
[error]                                               ^
[error] /home/gui/sbt-scalafix/src/main/scala/scalafix/sbt/ScalafixPlugin.scala:246:65: Illegal dynamic reference: loadedRules
[error]       val inputArgs = new ScalafixCompletions(workingDirectory, loadedRules).parser.parsed
[error]                                                                 ^
[error] two errors found
```

I tried to fix this by nesting `inputTaskDyn` and `taskDyn`. It always end up with an illegal dynamic reference.

Reply:
This is a dead-end: https://www.scala-sbt.org/1.0/docs/Input-Tasks.html#Constructing+the+Parser
>  Again, we were able to access the current Scala and sbt version for the project because they are settings. Tasks cannot be used to define the parser.

Reply:
Yes, I was surprised when you told me you wanted to provide completions for rewrites that are not built-in Scalafix, because that means you need to class load them and all that, which already brings you outside of the settings world.

Reply:
@MasseGuillaume why do you need to call a task? `scalafixAPI` is a setting https://github.com/scalacenter/sbt-scalafix/blob/7c00166740857df334b90c9a7b8701f66fca3721/src/main/scala/scalafix/sbt/ScalafixPlugin.scala#L237-L241